thesis-gui/src/components/my-quizzes/MyQuizzes.tsx

```diff
@@ -1,77 +1,98 @@
-import React from "react";
+import React, {useEffect, useState} from "react";
 import {
     IonButton,
-    IonContent,
+    IonContent, IonIcon,
     IonItem,
     IonLabel,
     IonList,
-    IonPage
+    IonPage, IonText, useIonToast
 } from "@ionic/react";
+import {keyOutline} from "ionicons/icons";
 import {PageHeader} from "../common/PageHeader";
+import {quizzesService} from "../../services/quizzes-service";
+import {QuizzesList} from "../../types/quizzes-list";
 
 const PAGE_NAME = "My Quizzes";
 
-const QUIZZES = [
-    {
-        name: "Quiz 1",
-        questionsNumber: 5
-    },
-    {
-        name: "Quiz 2",
-        questionsNumber: 27
-    },
-    {
-        name: "Quiz 3",
-        questionsNumber: 8
-    },
-    {
-        name: "Quiz 4",
-        questionsNumber: 13
-    }
-];
-
 const QuizzesListItem: React.FC<{
     name: string,
-    questionsNumber: number
-}> = ({ name, questionsNumber }) => {
+    questionsNumber: number,
+    _key: string
+}> = ({ name, questionsNumber, _key }) => {
+    const [present] = useIonToast();
+
+    const presentToast = () => {
+        present({
+            message: `"${name}" quiz key was copied to clipboard!`,
+            duration: 1500,
+            position: 'bottom'
+        });
+    };
+
+    const copyKeyToClipboard = () => {
+        navigator.clipboard.writeText(_key);
+        presentToast();
+    }
+
     return (
-        <IonItem button detail={true}>
+        <IonItem>
             <IonLabel>
                 <h2>{ name }</h2>
                 <p>{ `${questionsNumber} question(s)` }</p>
             </IonLabel>
+            <IonIcon
+                className="btn"
+                icon={keyOutline}
+                onClick={ copyKeyToClipboard }
+                style={{ paddingRight: "10px" }}
+            ></IonIcon>
         </IonItem>
     );
 }
 
-const QuizzesList: React.FC = () => {
-    const quizzesListItems = QUIZZES.map(quiz => (
-            <QuizzesListItem
-                name={ quiz.name }
-                questionsNumber={ quiz.questionsNumber }
-                key={quiz.name}
-            />
-        )
-    );
+export const MyQuizzes: React.FC = () => {
+    const [quizzesList, setQuizzesList] = useState<QuizzesList>([]);
+
+    const getQuizzesList = () => {
+        quizzesService.getList()
+            .then(res => setQuizzesList(res))
+            .catch(err => console.log(err));
+    }
+
+    useEffect(() => {
+        getQuizzesList();
+    }, []);
+
+    const quizzesListItems = quizzesList.map(quiz => {
+        return <QuizzesListItem
+            name={ quiz.questionsSetName }
+            questionsNumber={ quiz.numberOfQuestionsInSet }
+            _key={ quiz.questionsSetKey }
```

Comment:
There is an error in browser console that every list item needs a `key` property. Therefore, I would rename the custom property `_key` to something like `questionsSetKey` and add `key={quiz.questionsSetKey}` to avoid the error.